docs/getting-started/deployment-docker.md

```diff
@@ -0,0 +1,241 @@
+
+WORKDIR /app
+
+# Copy repo skeleton first, to avoid unnecessary docker cache invalidation.
+# The skeleton contains the package.json of each package in the monorepo,
+# and along with yarn.lock and the root package.json, that's enough to run yarn install.
+ADD yarn.lock package.json packages/backend/dist/skeleton.tar.gz ./
+
+RUN yarn install --frozen-lockfile --production --network-timeout 300000 && rm -rf "$(yarn cache dir)"
+
+# This will copy the contents of the dist-workspace when running the build-image command.
+# Do not use this Dockerfile outside of that command, as it will copy in the source code instead.
+ADD packages/backend/dist/bundle.tar.gz app-config.yaml ./
+
+CMD ["node", "packages/backend"]
+```
+
+For more details on how the `backend:bundle` command and the `skeleton.tar.gz`
+file works, see the
+
+The `Dockerfile` is typically placed at `packages/backend/Dockerfile`, but needs
+to be executed with the root of the repo as the build context, in order to get
+access to the root `yarn.lock` and `package.json`, along with any other files
+that might be needed, such as `.npmrc`.
+
+In order to speed up the build we can significantly reduce the build context
+size using the following `.dockerignore` in the root of the repo:
+
+```text
+.git
+node_modules
+packages
+!packages/backend/dist
+plugins
+```
+
+With the project build and the `.dockerignore` and `Dockerfile` in place, we are
+now ready to build the final image. Assuming we're at the root of the repo, we
+execute the build like this:
+
+```bash
+docker image build . -f packages/backend/Dockerfile --tag backstage
+```
+
+To try out the image locally you can run the following:
+
+```sh
+docker run -it -p 7000:7000 backstage
+```
+
+You should then start to get logs in your terminal, and then you can open your
+browser at `http://localhost:7000`
+
+[`backend:bundle` command docs](../cli/commands#backendbundle)
+
+### Multistage Build
+
+This section describes how to set up a multi-stage Docker build that builds the
+entire project within Docker. This is typically slower than a host build, but is
+sometimes desired because Docker in Docker is not available in the build
+environment or other requirements.
+
+The build is split into three different stages, where the first stage finds all
+of the `package.json`s that are relevant for the initial install step enabling
+us to cache the initial `yarn install` that installs all dependencies. The
+second stage executes the build itself, and is similar to the steps we execute
+on the host in the host build. The third and final stage then packages it all
+together into the final image, and is similar to the `Dockerfile` of the host
+build.
+
+The following `Dockerfile` executes the multi-stage build and should be added to
+the repo root:
+
+```Dockerfile
+# Stage 1 - Create yarn install skeleton layer
+FROM node:14-buster-slim AS packages
+
+WORKDIR /app
+COPY package.json yarn.lock ./
+
+COPY packages packages
```

Comment:
So this doesn't bust the cache for non-package-json changes? I actually thought the layers were "chained" like git commits essentially, such that a hash change here would taint the chain including after the rm below

Reply:
Yeah I'd imagine this works because the contents of the final image is hashed

Reply:
kinda surprised honestly